Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/ReportManager.kt

```diff
@@ -169,37 +179,97 @@ class ReportManager(
         Logger.d(TAG, "Stored new crash log, path = ${newCrashLog.absolutePath}")
     }
 
-    // Since this is a singleton we don't care about disposing of this thing because nothing may
-    // leak here
-    @SuppressLint("CheckResult")
-    fun sendCollectedCrashLogs() {
+    fun hasCrashLogs(): Boolean {
+        if (!createCrashLogsDirIfNotExists()) {
+            return false
+        }
+
+        val crashLogs = crashLogsDirPath.listFiles()
+        return crashLogs != null && crashLogs.isNotEmpty()
+    }
+
+    fun countCrashLogs(): Int {
+        if (!createCrashLogsDirIfNotExists()) {
+            return 0
+        }
+
+        return crashLogsDirPath.listFiles()?.size ?: 0
+    }
+
+    fun getCrashLogs(): List<File> {
         if (!createCrashLogsDirIfNotExists()) {
+            return emptyList()
+        }
+
+        return crashLogsDirPath.listFiles()
+                ?.sortedByDescending { file -> file.lastModified() }
+                ?.toList() ?: emptyList()
+    }
+
+    fun deleteCrashLogs(crashLogs: List<CrashLog>) {
+        if (!createCrashLogsDirIfNotExists()) {
+            settingsNotificationManager.cancel(SettingNotificationType.CrashLogs)
+            return
+        }
+
+        crashLogs.forEach { crashLog -> crashLog.file.delete() }
+
+        val remainingCrashLogs = crashLogsDirPath.listFiles()?.size ?: 0
+        if (remainingCrashLogs == 0) {
+            settingsNotificationManager.cancel(SettingNotificationType.CrashLogs)
             return
         }
 
+        // There are still crash logs left, so show the notifications if they are not shown yet
+        settingsNotificationManager.notify(SettingNotificationType.CrashLogs)
+    }
+
+    fun deleteAllCrashLogs() {
+        if (!createCrashLogsDirIfNotExists()) {
+            settingsNotificationManager.cancel(SettingNotificationType.CrashLogs)
+            return
+        }
+
+        val potentialCrashLogs = crashLogsDirPath.listFiles()
+        if (potentialCrashLogs.isNullOrEmpty()) {
+            Logger.d(TAG, "No new crash logs")
+            settingsNotificationManager.cancel(SettingNotificationType.CrashLogs)
+            return
+        }
+
+        potentialCrashLogs.asSequence()
+                .filter { file -> file.name.startsWith(CRASH_LOG_FILE_NAME_PREFIX) }
```

Comment:
Probably should remove this `filter` operator because only crash logs are supposed to be stored in that directory any way so everything should be deleted.